### PR TITLE
feat(zeta): add recovery functionality for zeta

### PR DESCRIPTION
**Pull request by Vijay-Jagannathan**
Ticket: WIN-296

### Context:

 - This PR adds the SDK changes necessary to recover a wallet
 - The main recover function lives inside the abstract cosmos module, so this PR will only have coin specific methods along with the necessary unit tests

### CODE REVIEW

File: modules/sdk-coin-zeta/test/unit/zeta.ts [hunk 1]
```diff
@@ -336,4 +342,131 @@ describe('Zeta', function () {
...
+    const testBalance = '15000000000000000';
+    const testAccountNumber = '123';
+    const testSequenceNumber = '0';
+    const testChainId = 'test-chain';
+
+    beforeEach(() => {
+      const accountBalance = sandBox.stub(Zeta.prototype, 'getAccountBalance' as keyof Zeta);
+      accountBalance.withArgs(wrwUser.senderAddress).resolves(testBalance);
+
+      const accountDetails = sandBox.stub(Zeta.prototype, 'getAccountDetails' as keyof Zeta);
```
Comment (DinshawKothari) [inline modules/sdk-coin-zeta/test/unit/zeta.ts:359]: can you mock the accountDetails response instead of stubbing it.
Check injective for reference.

Comment (Vijay-Jagannathan) [reply modules/sdk-coin-zeta/test/unit/zeta.ts:359]: sure, just to make sure we're calling the expected URL path and don't run into other errors ?

Comment (DinshawKothari) [reply modules/sdk-coin-zeta/test/unit/zeta.ts:359]: yeah correct.

File: modules/sdk-coin-zeta/src/lib/constants.ts [hunk 2]
```diff
@@ -2,3 +2,5 @@ export const validDenoms = ['azeta'];
 export const accountAddressRegex = /^(zeta)1(['qpzry9x8gf2tvdw0s3jn54khce6mua7l']{38})$/;
 export const validatorAddressRegex = /^(zetavaloper)1(['qpzry9x8gf2tvdw0s3jn54khce6mua7l']{38})$/;
 export const contractAddressRegex = /^(zeta)1(['qpzry9x8gf2tvdw0s3jn54khce6mua7l']+)$/;
+export const GAS_LIMIT = 250000;
+export const GAS_AMOUNT = '125000';
```
Comment (dpkjnr) [inline modules/sdk-coin-zeta/src/lib/constants.ts:6]: what is gas amount?

Comment (Vijay-Jagannathan) [reply modules/sdk-coin-zeta/src/lib/constants.ts:6]: gas fee for the transaction

Comment (dpkjnr) [reply modules/sdk-coin-zeta/src/lib/constants.ts:6]: nit: gas_amount looks confusing, fee_amount or fee_estimate would be better

**Review by dpkjnr — APPROVED** [review]

File: modules/sdk-coin-zeta/src/lib/constants.ts (hunk 2 above)
Comment (margueriteblair) [reply modules/sdk-coin-zeta/src/lib/constants.ts:6]: Also, what's the rationale for having one be a string and another one be a number between `GAS_LIMIT` AND `GAS_AMOUNT`

Comment (zahin-mohammad) [reply modules/sdk-coin-zeta/src/lib/constants.ts:6]:
```
export interface GasAmountDetails {
  gasAmount: string;
  gasLimit: number;
}
```

🙃

**Review by zahin-mohammad — APPROVED** [review]

File: modules/sdk-coin-zeta/src/lib/constants.ts (hunk 2 above)
Comment (Vijay-Jagannathan) [reply modules/sdk-coin-zeta/src/lib/constants.ts:6]: @margueriteblair 

this was intentional;
1. This is the common controller code uses which references the individual coin's `gasAmount` and `gasLimit` - https://github.com/BitGo/BitGoJS/blob/d5c6c74845e62ee3cd68321c55d729da1526fcae/modules/abstract-cosmos/src/cosmosCoin.ts#L171-L174
2. The `FeeData` type used above - https://github.com/BitGo/BitGoJS/blob/master/modules/abstract-cosmos/src/lib/iface.ts#L56-L59 contains mixed types and not just pure strings or pure numbers

File: modules/sdk-coin-zeta/package.json [hunk 3]
```diff
@@ -46,7 +46,8 @@
     "@cosmjs/amino": "^0.29.5",
     "@cosmjs/encoding": "^0.29.5",
     "@cosmjs/stargate": "^0.29.5",
-    "bignumber.js": "^9.1.1"
+    "bignumber.js": "^9.1.1",
+    "@bitgo/sdk-lib-mpc": "^8.5.0"
```
Comment (evanzbitgo) [inline modules/sdk-coin-zeta/package.json:50]: @Vijay-Jagannathan master branch is currently broken https://github.com/BitGo/BitGoJS/actions/runs/6015424750/job/16317283340

Comment (Vijay-Jagannathan) [reply modules/sdk-coin-zeta/package.json:50]: weird how it didn't show up in the [build](https://github.com/BitGo/BitGoJS/actions/runs/5998299142/job/16266306825?pr=3835). Also shouldn't it work for all minor versions ?

Comment (Vijay-Jagannathan) [reply modules/sdk-coin-zeta/package.json:50]: Being addressed in this PR - https://github.com/BitGo/BitGoJS/pull/3859

**Pull request merged**